### ISSUE TITLE
TypeError encountered in fit_transform during the reduction of topics stage when nr_topics="auto"

### ISSUE THREAD

**Issue by thisisclement** [MaartenGr/BERTopic#210]
Hello there! Not sure if you are already aware of this, but I would like to highlight to you just in case you havent. :) During the fit_transform reduction of topics stage, following error was encountered:
`TypeError: A sparse matrix was passed but dense data is required. Use X.toarray() to convert to a dense numpy array`
Without the the `nr_topics="auto"` it ran without issues.  
I am running `v0.8.1` and i reproduced it with `v0.9.0` too. 

This was the code I was running:
```python
model = SentenceTransformer("paraphrase-multilingual-MiniLM-L12-v2")
embeddings = model.encode(docs, show_progress_bar=True)
topic_model = BERTopic(verbose=True, calculate_probabilities=True, \
                       n_gram_range=(1,3), low_memory=True, nr_topics="auto")
topics, probs = topic_model.fit_transform(docs, embeddings=embeddings)
```

Have you encountered this before? Thanks!  

<details>
<summary>Here is the full stack trace:</summary>

```python
TypeError                                 Traceback (most recent call last)
<ipython-input-18-465c074474bc> in <module>
----> 1 topics, probs = topic_model.fit_transform(docs, embeddings=embeddings)

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/bertopic/_bertopic.py in fit_transform(self, documents, embeddings, y)
    291         # Reduce topics
    292         if self.nr_topics:
--> 293             documents = self._reduce_topics(documents)
    294             probabilities = self._map_probabilities(probabilities)
    295         predictions = documents.Topic.to_list()

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/bertopic/_bertopic.py in _reduce_topics(self, documents)
   1509                 documents = self._reduce_to_n_topics(documents)
   1510         elif isinstance(self.nr_topics, str):
-> 1511             documents = self._auto_reduce_topics(documents)
   1512         else:
   1513             raise ValueError("nr_topics needs to be an int or 'auto'! ")

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/bertopic/_bertopic.py in _auto_reduce_topics(self, documents)
   1590                                       metric='euclidean',
   1591                                       cluster_selection_method='eom',
-> 1592                                       prediction_data=True).fit_predict(norm_data[1:])
   1593 
   1594         # Map similar topics

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/hdbscan/hdbscan_.py in fit_predict(self, X, y)
    939             cluster labels
    940         """
--> 941         self.fit(X)
    942         return self.labels_
    943 

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/hdbscan/hdbscan_.py in fit(self, X, y)
    920 
    921         if self.prediction_data:
--> 922             self.generate_prediction_data()
    923 
    924         return self

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/hdbscan/hdbscan_.py in generate_prediction_data(self)
    962                 self._raw_data, self.condensed_tree_, min_samples,
    963                 tree_type=tree_type, metric=self.metric,
--> 964                 **self._metric_kwargs
    965             )
    966         else:

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/hdbscan/prediction.py in __init__(self, data, condensed_tree, min_samples, tree_type, metric, **kwargs)
    100         self.raw_data = data.astype(np.float64)
    101         self.tree = self._tree_type_map[tree_type](self.raw_data,
--> 102                                                    metric=metric, **kwargs)
    103         self.core_distances = self.tree.query(data, k=min_samples)[0][:, -1]
    104         self.dist_metric = DistanceMetric.get_metric(metric, **kwargs)

sklearn/neighbors/_binary_tree.pxi in sklearn.neighbors._kd_tree.BinaryTree.__init__()

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     61             extra_args = len(args) - len(all_args)
     62             if extra_args <= 0:
---> 63                 return f(*args, **kwargs)
     64 
     65             # extra_args > 0

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/sklearn/utils/validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)
    651                                       dtype=dtype, copy=copy,
    652                                       force_all_finite=force_all_finite,
--> 653                                       accept_large_sparse=accept_large_sparse)
    654     else:
    655         # If np.array(..) gives ComplexWarning, then we convert the warning

~/anaconda3/envs/bertopic-new/lib/python3.7/site-packages/sklearn/utils/validation.py in _ensure_sparse_format(spmatrix, accept_sparse, dtype, copy, force_all_finite, accept_large_sparse)
    415 
    416     if accept_sparse is False:
--> 417         raise TypeError('A sparse matrix was passed, but dense '
    418                         'data is required. Use X.toarray() to '
    419                         'convert to a dense numpy array.')

TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.
```
</details>

**Comment by thisisclement** [comment]
Thanks for working on it! Glad I managed to surface it out for you to tag it to the next release! 
Also, interesting to note is that when I ran instantiated BERTopic with an embedding model like below, it runs without any issues. 

```python
topic_model = BERTopic(model=model, verbose=True, calculate_probabilities=True, \
                       n_gram_range=(1,3), low_memory=True, nr_topics="auto")
```

**Comment by MaartenGr** [comment]
That was actually to be expected. When you use custom embeddings and do not specify an embedding model, no embedding model will be used at all during training in BERTopic. The resulting topic vectors will then be created using the c-TF-IDF matrix instead of the embedding model's topic representation. The difference between the two is that the former results in a sparse matrix whereas the latter doesn't. This explains the issue you are having.

**Comment by MaartenGr** [comment]
A fix for this issue has been released (v0.9.1). To use this fix, update BERTopic with: `pip install --upgrade bertopic`. If there any other issues you are experiencing, please let me know!
